### ISSUE TITLE
GUNNEXP2

### ISSUE THREAD

**Issue by jensroes** [jensroes/prowrite-mixture-models#1]
@Mark-Torrance just need some clarification on variables:

- `ispwordinit` was just used to identify word initial transitions and can be ignored?
- `pwd_isfluent`: I guess we just need the fluent ones but what does `NA` mean here?
- `topic` was used to counterbalancing the `xn` condition I assume. I'd include `topic` as fixed effect.

**Comment by Mark-Torrance** [comment]
> * `ispwordinit` was just used to identify word initial transitions and can be ignored?

Yes. I'm not sure whether this simply tags `a._^a` or if there was more to it (I suspect not). But, yes, ignore.

> * `pwd_isfluent`: I guess we just need the fluent ones but what does  `NA` mean here?

Without checking, you just get `NA` for space-bar presses, right? Which aren't part of a word.
 
> * `topic` was used to counterbalancing the  `xn` condition I assume. I'd include  `topic` as fixed effect.

Yes. I'd normally include this as a random effect. Were you planning a full-factorial design including topic? I can see (I think) why you might want to include as a fixed effect, but I'm afraid that this will confuse things (by which I mean that I really don't want to find that there's an interaction with topic :).  There is also order, which was also counterbalanced. I need to check whether I know order, and we should include that too (again as a random effect, I suggest). I can't easily identify order in Gunn's data, though (I don't think).

**Comment by jensroes** [comment]
Some initial results [here](https://rpubs.com/jensroes/991090). I've combined it with he comparison to other data sets, run model comparisons, and calculated the masking effects (for the unconstrained model). However,

you asked me to run a model where the short component doesn't distinguish masked and unmasked. I did that and one chain is done but the other chains haven't even started yet. This indicates that the model struggles to explore the posterior and I can't see why. So I think this models isn't a good idea but I'll keep working on it. I'm actually surprised that this model is struggling. I'll also try it with other data.

I've also fitted a model that doesn't distinguish the short component for transition locations or masking. This model has a lower predictive performance than the unconstrained model. Therefore I haven't included the masking effect for the constrained model (at least not yet).

So I think the unconstrained model is the best choice here (for now) and for the other data sets. What the posterior shows us is that masking leads to longer pre-sentence pauses but also to longer intervals that are associated with fluent typing (the short component). The way I think about this is that people write overall slower when writing masked text.

Aside, at the moment I'm completely ignoring "topic" in the model because I'm not sure we need it at all.

**Comment by Mark-Torrance** [comment]
The results for the unconstrained model make a lot of sense to me. They're consistent with a theory that 

(a) lookback is an automatised response to not being sure what to say next
(b) what writers look at when they lookback serves to prompt what to say next

So in the masked condition they are looking back but then finding this doesn't help them, so are slower to retrieve the message (or syntax, or whatever the level at which self-priming is taking place) that they need for the next sentence. Bear in mind that these are all non-editing transitions.

We actually have some eye data (Gunn experiment 1) which suggests that they are looking back just as much in the masked condition as in the normal condition.

**Comment by Mark-Torrance** [comment]
You say that you haven't completed a run of the constrained model. But 7.1.3 reports a constrained model. So what am I missing?

I was set to argue strongly that *a priori* the constrained model is the right way to go. But, partly because I've now see your results, I'm not sure that's correct. For the parameter estimates in the constrained model (7.1.3) what was the short component duration? I'm assuming it was greater than the mean of the short component duration in the unconstrained model?

**Comment by jensroes** [comment]
> You say that you haven't completed a run of the constrained model. But 7.1.3 reports a constrained model. So what am I missing?

There are two constrained models. (1) 1 short component across transitions locations and masking condition (2) 1 short component per transition location across masking conditions. Model (1) works, but model (2) what you asked for doesn't make a move.

For the constrained model the short component has a mean of 157 msecs (PI: 147, 168) which is ever so slightly longer than the within word transitions of the unconstrained model.

> I'm assuming it was greater than the mean of the short component duration in the unconstrained model?

Yes!

> We actually have some eye data (Gunn experiment 1) which suggests that they are looking back just as much in the masked condition as in the normal condition.

Should I mixture model those too?

**Comment by Mark-Torrance** [comment]
> (1) 1 short component across transitions locations and masking condition 

I can't remember why I thought (2). This is the one that makes sense (possibly).

> For the constrained model the short component has a mean of 157 msecs (PI: 147, 168) which is ever so slightly longer than the within word transitions of the unconstrained model.

I don't think there's a simple answer to which model you think is right. This depends entirely on what you want to short (and therefore long) IKIs to represent. For the argument that I want to develop, I think the fully constrained model (constant across location and across condition) is what I need. But good to have both.

**Comment by jensroes** [comment]
> I can't remember why I thought (2). This is the one that makes sense (possibly).

(2) makes sense to me and does easily converge for one of the other data sets. (1) gives divergent transitions from time to time but it isn't really that bad. For transitions durations but not lookback durations I get much better fits for the unconstrained model. The constrained model seems to be better for lookback durations (by which I mean the models have a similar predictive performance, hence the simpler model, ie. the constrained model, is more parsimonious.

Hence,

> I don't think there's a simple answer to which model you think is right. This depends entirely on what you want to short (and therefore long) IKIs to represent.

on the basis of model comparisons we can conclude that the unconstrained model captures the transition durations better and is therefore a better model of the cognitive process that generates keystrokes.

> This depends entirely on what you want to short (and therefore long) IKIs to represent. For the argument that I want to develop, I think the fully constrained model (constant across location and across condition) is what I need.

I agree. I think we briefly discussed this before but I still haven't accepted (by which I mean understood your argument). In my view the unconstrained model is better for keystrokes because it doesn't just allow that sometimes pauses are longer or pauses are more frequent but also that under some conditions people execute keystrokes more slowly which is captured by the short component). That we don't see this in lookback durations might be because of how eyes move but also because the short distribution of lookback durations is overall longer.

**Comment by Mark-Torrance** [comment]
Jens, could you share the posteriors for the constrained model (i.e. the file that corresponds to this)...

```R
file <- list.files(path = "./prowrite-mixture-models-main/stanout/gunnexp2", 
                   pattern = "mogbetac.*.csv$",
                   full.names = T)
```

 I just want the option of replotting for my Potsdam talk.

**Comment by jensroes** [comment]
[mogbetaconstr.zip](https://github.com/jensroes/prowrite-mixture-models/files/11908506/mogbetaconstr.zip)

Let me know if anything doesn't make sense. 

Beta is the short component, beta2 is long component and delta is their difference. theta is the mixing proportion, so the probability of beta2, on the logit scale and prob is theta as probability. 

For a tutorial what you can do with posteriors like this, check here: https://rpubs.com/jensroes/1000459 although I think there isn't anything new to you in this.

**Comment by Mark-Torrance** [comment]
Brill. Thanks. I'm enjoying looking at (and learning from) your code - though I've not explored the Stan bits yet (and don't need to).

**Comment by jensroes** [comment]
I can download it. Its 8MB zipped and 35 unzipped. I'll email it to you

**Comment by Mark-Torrance** [comment]
Working for me now too.
